### PR TITLE
[PAY-458] Buy Audio Modal Input Page

### PR DESCRIPTION
**Pull request by rickyrombo**
### Description

Adds the input page to the Buy Audio Modal

image
image
image

(Ignore weird amounts in screenshot, my root SOL balance is non-zero making it look funky)

Changes not included but tracked separately:
- Update Coinbase Pay button for dark mode
- Update Radio Pill Button to not have focus state
- Add padding for Coinbase fees

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

❗ Puts CoinbasePayProvider at root of app

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

Tested locally pointed at stage (and prod)

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

Already behind BUY_AUDIO_ENABLED feature flag

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-buy-audio-input

File: packages/web/src/components/buy-audio-modal/BuyAudioModal.tsx [hunk 1]
```diff
@@ -1,11 +1,12 @@
-import React, { useCallback } from 'react'
+import { useCallback } from 'react'
 
 import { Modal, ModalContent, ModalHeader, ModalTitle } from '@audius/stems'
```
Comment (dylanjeffers) [inline packages/web/src/components/buy-audio-modal/BuyAudioModal.tsx:3]: I think we might have the icon for solana already, we are using it in the collectibles list items i believe.

File: packages/web/src/components/buy-audio-modal/components/AudioAmountPicker.tsx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+          rightLabel={'$AUDIO'}
+          value={customAmount}
+          isNumeric={true}
+          isWhole={true}
+          onChange={handleCustomAmountChange}
+        />
+      ) : null}
+    </>
+  )
+}
```
Comment (dylanjeffers) [inline packages/web/src/components/buy-audio-modal/components/AudioAmountPicker.tsx:142]: this. file is sooo clean

File: packages/web/src/components/buy-audio-modal/components/AudioAmountPicker.tsx [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+      </RadioButtonGroup>
+      {isCustomAmountInputVisible ? (
+        <TokenValueInput
+          rightLabelClassName={styles.customAmountLabel}
+          inputClassName={styles.customAmountInput}
+          format={Format.INPUT}
+          placeholder={'Enter an amount'}
+          rightLabel={'$AUDIO'}
+          value={customAmount}
+          isNumeric={true}
```
Comment (dylanjeffers) [inline packages/web/src/components/buy-audio-modal/components/AudioAmountPicker.tsx:135]: personally like `isNumeric` vs `isNumeric={true}` same is true for strings, `rightLabel='$AUDIO'`, but all good, ideally we have linter autofix everything

**Review by dylanjeffers — APPROVED** [review]
such clean code, nice work

File: packages/web/src/components/buy-audio-modal/BuyAudioModal.tsx (hunk 1 above)
Comment (rickyrombo) [reply packages/web/src/components/buy-audio-modal/BuyAudioModal.tsx:3]: I think that one is slightly different!

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-buy-audio-input

File: packages/web/src/components/buy-audio-modal/components/PurchaseQuote.tsx [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    if (purchaseInfo.errorType === PurchaseInfoErrorType.MAX_AUDIO_EXCEEDED) {
+      errorMessage = messages.aboveMaximumError(purchaseInfo.maxAudio)
+    } else if (
+      purchaseInfo.errorType === PurchaseInfoErrorType.MIN_AUDIO_EXCEEDED
+    ) {
+      errorMessage = messages.belowMinimumError(purchaseInfo.minAudio)
+    } else {
+      errorMessage = messages.unknownError
+    }
+  }
```
Comment (piazzatron) [inline packages/web/src/components/buy-audio-modal/components/PurchaseQuote.tsx:78]: nit: could just index into an object mapping from error message -> message for this block

**Review by piazzatron — APPROVED** [review]
Code and UI looks fantastic! Only thing I noticed in testing is that we should probably disable the buy button on the modal until we've input an amount and have a quote?

File: packages/web/src/components/buy-audio-modal/components/PurchaseQuote.tsx (hunk 4 above)
Comment (rickyrombo) [reply packages/web/src/components/buy-audio-modal/components/PurchaseQuote.tsx:78]: can we easily do that though? Two of them are functions that take different params.

**Pull request merged**